src/components/LinkEncoder.js

```diff
@@ -91,29 +99,50 @@ const LinkEncoder = () => {
 
     const [buttonText, setButtonText] = useState('Connect');
     const [selected, setSelected] = useState("success");
+    const [checked, setIsChecked] = useState(false);
 
     const connectAndDisconnect = async (e) => {
-        e.preventDefault();
-
-        await window.linkEncoderAPI.connectionLinkEncoder(postData.ip, postData.port);
+        
+
+        await window.linkEncoderAPI.connectionLinkEncoder(postData.ip, postData.port)
+        .then(() => {
+            if (buttonText == 'Connect'){
+                setButtonText('Disconnect');
+                setSelected("error");
+            }
+            else {
+                setButtonText('Connect');
+                setSelected("success");
+            }  
+        });
     };
 
-    const handleClick = () => {
-        if (buttonText == 'Connect'){
-            setButtonText('Disconnect');
-            setSelected("error");
-        }
-        else {
-            setButtonText('Connect');
-            setSelected("success");
-        }   
+    const pinging = async () => {
+        return new Promise(resolve => {
+            if (checked == false) {
+                interval = setInterval(async () => {
+                    
+                    var today = new Date();
+                    var time = today.getHours() + ":" + today.getMinutes() + ":" + today.getSeconds();
+                    await window.linkEncoderAPI.sendToLinkEncoder(time);    
+                }, 5000);
+                
+            }
+            resolve();
+        })
     };
 
+    const stopPinging = async () => {
+        setIsChecked(!checked);
+        clearInterval(interval);
```

Comment:
![image](https://user-images.githubusercontent.com/90006274/198724703-1e666381-95c3-4c1e-bc67-5311ac888556.png)
![image](https://user-images.githubusercontent.com/90006274/198724769-808b0cb8-62a9-42ff-8809-eca75e3484be.png)
If interval is not a return value of setInterval method, then it will do nothing.